apps/src/templates/projects/projectsModule.js

```diff
@@ -15,7 +21,52 @@ export function selectedGallery(state, action) {
   }
 }
 
+export function projectLists(state, action) {
+  // A map from project type to array of projects, e.g.:
+  //   {
+  //     applab: [{...}, {...}, {...}]
+  //     gamelab: [{...}, {...}, {...}]
+  //     playlab: [{...}, {...}, {...}]
+  //     artist: [{...}, {...}, {...}]
+  //   }
+  state = state || {};
+  switch (action.type) {
+    case SET_PROJECT_LISTS:
+      return action.projectLists;
+    case ADD_OLDER_PROJECTS: {
+      // Append the incoming list of older projects to the existing list,
+      // removing duplicates.
+      const {projects, projectType} = action;
+      state = {...state};
+      state[projectType] = _.unionBy(state[projectType], projects, 'channel');
+      return state;
```

Comment:
I'm open to using Immutable -- AFAIK the benefit is that you avoid the trap whereby you might alter the redux internal state and then return the same altered state object, thereby causing redux to not notice that the state has changed. However my understanding gets a little fuzzy with what happens if the Immutable contains references to mutable arrays and then the reducer modifies those arrays. Do you have a sense of what has to happen here to avoid this trap? Do the sub-arrays need to be created via Immutable also?